doc/system_description/software.md

```diff
@@ -255,188 +256,6 @@ Then we continue with the CDI computation by updating with an optional
 USS and then finalizing the hash, storing the resulting digest in
 `CDI`.
 
-### Firmware protocol
```

Comment:
With these changes, should we remove the note on the top of the document?
Or what do we foresee that our documentation should live?

Right now we have three; here, close to the cores and dev.tillitis. Should we keep all?

Reply:
Might be a bigger discussion than this specific PR.

Reply:
I think the direction things are going according to earlier chats is that we move specific documentation closer to the thing it documents, so I think the firmware parts of `software.md` should move to a README in the firmware directory, for instance.

Do we do that in this branch? What do you think?

Reply:
Great, I agree. Documentation close to the source makes it easier for the developer and the one trying to find it. 

No, let's do it in multiple smaller branches and PRs. Makes it easier than one big.